go/apps/dialogue/tests/test_vumi_app.js

```diff
@@ -318,5 +320,68 @@ describe("app", function() {
                     });
             });
         });
+
+        describe("when the user enters a send state which is a httpjson type", function() {
+            it("should successfully fetch the data from the url if the method is GET", function() {
+                return tester
+                    .setup(function(api) {
+                        api.contacts.add({
+                           msisdn: '+27123'
+                        });
+                        api.http.fixtures.add({
+                            request: {
+                                method: "GET", 
+                                url: "www.foo.bar"
+                            }
+                        });
+                    })
+                    .setup.user.addr('+27123')
+                    .setup.user.state('choice-1')
+                    .input('4')
+                    .check(function(api) {
+                        assert.equal(api.http.fixtures.fixtures[0].responses[0].code, 200);
+                    })
+                    .run();
+            });
+            
+            it("should successfully send the data to the url if the method is POST", function() {
+                return tester
+                    .setup.user.addr('+27123')                    
+                    .setup.user.state('choice-1')
+                    .setup(function(api) {
+                        api.contacts.add({
+                            msisdn: '+27123'
+                        });
+
+                        api.contacts.store[0].extra = { 
+                            'message-1': 'value-5', 
+                            'message-1-1': 'value-5' 
+                        };
+
+                        api.http.fixtures.add({
+                            request: {
+                                method: "POST", 
+                                url: "www.foo.bar", 
+                                data: JSON.stringify({
+                                    user: {
+                                        answers: { 
+                                            'choice-1': 'value-5'
+                                        }
+                                    }, 
+                                    contact: api.contacts.store[0], 
+                                    conversation_key: poll.conversation_key
+                                })
+                            }
+                        });
+
+                        api.contacts.store[0].extra = {};
```

Comment:
extra is empty when the fixture is setup, if I pre-populate during the fixture and not empty it after the fixture has been created the choise adds two more entries to extra and then the test fails.